src/Spectre.Data/Datasets/BasicTextDataset.cs

```diff
@@ -0,0 +1,307 @@
+            _mz = mz;
+            AppendFromRawData(data);
+        }
+
+        /// <summary>
+        /// Method parsing the whole text file and initializing the array
+        /// with found data.
+        /// </summary>
+        /// <param name="filePath">Path to the text file.</param>
+        /// <exception cref="InvalidDataException">Thrown where there is a problem with file parsing.</exception>
+        public void AppendFromFile(string filePath)
+        {
+            //TODO: Specifying formalized format of data in text files.
+            //TODO: Safety check, format check
+            try
+            {
+                using (StreamReader sr = new StreamReader(filePath))
+                {
+                    sr.ReadLine(); // omit global metadata
+                    sr.ReadLine(); // omit m/z values
+                    while (sr.Peek() > -1)
+                    {
+                        var metadata = sr.ReadLine()?.Split(null);
+                        var intensities = sr.ReadLine()?.Split(null);
+                        if (metadata == null || intensities == null)
+                            continue;
+                        if (intensities.Length != _mz.Length)
+                            throw new InvalidDataException("Length of the data must be equal to length of m/z values.");
+
+                        int x, y, z;
+
+                        if (!int.TryParse((metadata.Length>0)?metadata[0]:null, NumberStyles.Any, CultureInfo.InvariantCulture, out x))
+                            x = -1;
+                        if (!int.TryParse((metadata.Length>1)?metadata[1]:null, NumberStyles.Any, CultureInfo.InvariantCulture, out y))
+                            y = -1;
+                        if (!int.TryParse((metadata.Length>2)?metadata[2]:null, NumberStyles.Any, CultureInfo.InvariantCulture, out z))
+                            z = -1;
+
+                        _spatialCoordinates.Add(new SpatialCoordinates(x, y, z));
+                        _intensity.Add(new double[_mz.Length]);
+
+                        int backIdx = _intensity.Count - 1;
+                        for (int i = 0; i < intensities.Length; i++)
+                            if (!double.TryParse(intensities[i], NumberStyles.Any, CultureInfo.InvariantCulture, out _intensity[backIdx][i]))
+                                _intensity[backIdx][i] = double.NaN;
+                            
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                throw new InvalidDataException("Error while parsing " + filePath + " file.", e);
+            }
+        }
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        /// <exception cref="InvalidDataException">Throws when the data is null or 
+        /// the length of data is not matching the dataset length.</exception>
+        public void AppendFromRawData(double[,] data)
```

Comment:
I think, in the next iteration there should be at least optional argument to add `SpatialCoordinates` for appended data.